src/main/java/org/truffleruby/core/basicobject/BasicObjectNodes.java

```diff
@@ -126,11 +126,16 @@ public boolean equal(DynamicObject a, DynamicObject b) {
             return a == b;
         }
 
-        @Specialization(guards = { "isNotDynamicObject(a)", "isNotDynamicObject(b)", "notSameClass(a, b)", "isNotIntLong(a) || isNotIntLong(b)" })
+        @Specialization(guards = { "isNotDynamicObject(a)", "isNotDynamicObject(b)", "!sameClass(a, b)", "isNotIntLong(a) || isNotIntLong(b)" })
```

Comment:
I don't have a problem with the change per se, but it looks a bit weird when the four other methods called in the guards have an "isNot" prefix.